### DISCUSSION TITLE
[Feature] Image Details: GPS Altitude

### DISCUSSION THREAD

**Discussion by jtamm-red** [immich-app/immich#6915] [Feature Request]
### The feature

During my mountain hiking adventure, I captured a stunning photograph using my mobile device, which was seamlessly uploaded through the Immich app. The captivating scenery brought me immense joy, and now I am eager to discover the altitude of the location.

Upon attempting to retrieve this information using exiftool, I found that the image contains GPS data.
Executed command `exiftool IMG_20240114_115349.jpg | grep GPS`.
```
GPS Latitude Ref                : North
GPS Longitude Ref               : East
GPS Altitude Ref                : Above Sea Level
GPS Time Stamp                  : 10:53:48
GPS Processing Method           : GPS
GPS Date Stamp                  : 2024:01:14
GPS Altitude                    : 3400.8 m Above Sea Level
GPS Date/Time                   : 2024:01:14 10:53:48Z
GPS Latitude                    : 46 deg 4' 48.14" N
GPS Longitude                   : 7 deg 19' 6.02" E
GPS Position                    : 46 deg 4' 48.14" N, 7 deg 19' 6.02" E
```
To extract the altitude value programmatically, we can use exiftool and parse the result as an array. The first element of the array represents the numerical value, while the second element indicates the unit type in string format.

I recommend displaying the GPS Altitude information in the detailed image information for a more comprehensive overview.
![image](https://github.com/immich-app/immich/assets/53046846/592c89ef-6694-48b2-a71d-07786c1356c0)

### Platform

- [x] Server
- [X] Web
- [X] Mobile

**Comment by T0biasCZe** [comment]
Besides the Altitude, it could also display the GPS Direction (compass), which EXIF also includes: https://exiftool.org/TagNames/GPS.html            
GPSImgDirection is the angle in degrees, 0-359.99, relative to either magnetic north or true north (set in GPSImgDirectionRef) (though some cameras round it to nearest 45°)
eg something like this:
![image](https://github.com/user-attachments/assets/e970109c-31b7-40b9-b6df-346d8135f6e1)



and then, theoretically, it could also display the atmospheric pressure, but that one is highly debatable since the EXIF tag is not standardized and different manufacturers have different tag name, so if immich was to include it, each of the potential tag names would have to be supported...

**Comment by joernex** [comment]
Perhaps it would be better suited as a separate feature request, but I would like to expand on this really good suggestion.

While latitude and longitude are already used in many ways, I suggest that altitude should not only be displayed in the details of individual images. It should also be a property that can be filtered on in search and on the map.

For example, filter options like
- a checkbox to exclude images without altitude
	- unchecked by default
	- maybe auto-adjusted in the search: checked when a first range is set, ~~unchecked when the last range is removed~~ (this would prevent including irrelevant no-value images during range filtering)
- and input fields (min/max or open) or a reasonable slider for adding a range as limitation
	- supporting several ranges at the same time (OR)
	- no selected range corresponds to the entire value range

Use cases
- Find images in a defined altitude range using the search (e.g. alpine and above).
- Set a maximum value on the map to hide images taken at the typical cruising altitude of aircraft. This removes "noise" from the map in places where you have  _not actually_ been.

I would find this wonderful.

**Comment by prabinlamsal19** [comment]
@alextran1502  
I would like to work on this issue. Are we positive about this being merged/needed? : )
